llvm/lib/ObjectYAML/DXContainerEmitter.cpp

```diff
@@ -219,6 +219,25 @@ void DXContainerWriter::writeParts(raw_ostream &OS) {
             El.Allocated, El.Kind, El.Type, El.Mode, El.DynamicMask,
             El.Stream});
 
+      for (int I = 0; I < 4; ++I) {
```

Comment:
If you do change to using `std::array` as my other comment suggests, it'd be good to keep a static assert here:
```c++
      static_assert(PSV.OutputVectorMasks.size() == PSV.InputOutputMap.size());
      for (size_t I = 0, E = PSV.OutputVectorMasks.size(); I != E; ++I) {
        ...
```